packages/styled-components/src/models/StyledComponent.js

```diff
@@ -261,6 +277,10 @@ export default function createStyledComponent(
   });
 
   if (process.env.NODE_ENV !== 'production') {
```

Comment:
So this is probably just a mildly controversial comment but I think we've previously been a bit heavy on warnings.

Yes, they may catch potential issues early, but at some point we'll have to think of the likelihood of them occurring.

Do we really think that this is going to provide value over a stack trace that is already clearly pointing to the same interpolation that could be causing this issue when a theme was missing?

Reply:
There will only be a stack trace if they access a deeper property like `theme.colors.red`. We backfill `theme` with an empty object so `theme.color` wouldn't trigger a hard error, it'd just be undefined in the styles so the console error definitely would help catch bugs with more subtle usage.